pkg/argocd/argocd.go

```diff
@@ -429,6 +431,16 @@ func GetImagesFromApplication(app *v1alpha1.Application) image.ContainerImageLis
 		for _, img := range *parseImageList(updateImage) {
 			if forceStr, force := annotations[fmt.Sprintf(common.ForceUpdateOptionAnnotation, img.ImageAlias)]; force && strings.ToLower(forceStr) == "true" {
 				if images.ContainsImage(img, false) == nil {
+					// code consuming this list assumes the tag is valid, but for a force update, we may only have a constraint which isn't a valid tag
+					// if using semver, try to parse the tag and remove it if it fails
+					if strategy, set := annotations[fmt.Sprintf(common.UpdateStrategyAnnotation, img.ImageAlias)]; (!set ||
+						image.ParseUpdateStrategy(strategy) == image.VersionSortSemVer) && img.ImageTag != nil {
+						_, err := semver.NewVersion(img.ImageTag.TagName)
+						if err != nil {
+							log.Debugf("Dropping tag from force-update of image %s because %v", img, err)
+							img.ImageTag = &tag.ImageTag{}
+						}
+					}
```

Comment:
I believe we can simply set the tag to `nil` from what is specified in `image-list`, without having to perform checks for the update strategy or whether the tag can be correctly parsed as a semver.

If an image tag is `nil` in one of the images we check for, we explicitly create an empty tag [here](https://github.com/argoproj-labs/argocd-image-updater/blob/93d76f91a8242421edf6300cabf86c9dac1ac028/pkg/argocd/update.go#L160) and handle that case appropriately [here](https://github.com/argoproj-labs/argocd-image-updater/blob/93d76f91a8242421edf6300cabf86c9dac1ac028/pkg/image/version.go#L86)

So I think we can safely remove all the logic here, and just set `img.ImageTag = nil` here.

Reply:
Makes sense! I'll update shortly